src/LibraryManager/CacheService/CacheService.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Web.LibraryManager.Contracts;
+
+namespace Microsoft.Web.LibraryManager
+{
+    /// <summary>
+    /// Service to manage basic operations on libraries cache
+    /// </summary>
+    internal class CacheService
+    {
+        // TO DO: Move these expirations to the provider 
+        private readonly int _catalogExpiresAfterDays = 3;
+        private readonly int _libraryExpiresAfterDays = 1;
+        private IWebRequestHandler _requestHandler;
+
+        public CacheService(IWebRequestHandler requestHandler)
+        {
+            _requestHandler = requestHandler;
+        }
+
+        /// <summary>
+        /// Deletes files from cache 
+        /// </summary>
+        public void ClearCache()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Returns the provider's catalog from the provided Url to cacheFile
+        /// </summary>
+        /// <param name="url"></param>
+        /// <param name="cacheFile"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public async Task<string> GetCatalogAsync(string url, string cacheFile, CancellationToken cancellationToken)
```

Comment:
All these methods at the very least need to verify that the cacheFile is inside the cache root directory and not some arbitrary location.

Reply:
CacheService will move out of the provider so we own the location the files will be saved. These changes should be in the extensibility PR